libs/feature/search/src/lib/favorites/favorite-star/favorite-star.component.spec.ts

```diff
@@ -115,54 +115,90 @@ describe('FavoriteStarComponent', () => {
       describe('if record is not part of favorite', () => {
         beforeEach(() => {
           ;(favoritesService as any).myFavoritesUuid$.next([
-            'aaa',
-            'bbb',
-            'ccc',
+            component.record.uuid,
           ])
-          starToggle.newValue.emit(true)
           fixture.detectChanges()
         })
-        it('adds record to favorites', () => {
-          expect(favoritesService.addToFavorites).toHaveBeenCalledWith([
-            component.record.uuid,
-          ])
-          expect(favoritesService.removeFromFavorites).not.toHaveBeenCalled()
+        describe('if change is made from this component', () => {
+          beforeEach(() => {
+            starToggle.newValue.emit(true)
+            fixture.detectChanges()
+          })
+          it('adds record to favorites', () => {
+            expect(favoritesService.addToFavorites).toHaveBeenCalledWith([
+              component.record.uuid,
+            ])
+            expect(favoritesService.removeFromFavorites).not.toHaveBeenCalled()
+          })
+          it('increase record favorite count by one', () => {
+            expect(favoriteCountEl.textContent).toEqual(
+              (component.record.favoriteCount + 1).toString()
+            )
+          })
+          it('adds the record to myFavoritesUuid$', async () => {
+            await expect(favoritesService.myFavoritesUuid$['_value'][0]).toBe(
```

Comment:
I don't understand what does this test actually test.
BTW, Observables are a flow and should not be tested as an instantaneous value (which is hacky cause it's not the public API).

Reply:
It tests if the record is present in the `myFavoriteUuid` after the user clicks on the star. It's important especially when it ticks from another component.

Reply:
Okay, so how do I test if `myFavoriteUuid` properly recorded that the record is in the user's favorite? I didn't have any other idea

Reply:
`component.record.uuid = 'a'`

- `myFavoriteUuid$.next(['b'])` => nothing happens
- `myFavoriteUuid$.next(['a'])` => the component is updated

Reply:
> It tests if the record is present in the `myFavoriteUuid` after the user clicks on the star. It's important especially when it ticks from another component.

yes but you did the `myFavoritesUuid$.next([component.record.uuid])` in the `beforeEach`, which is exactly what you test in there.